test/test_optim.py

```diff
@@ -236,6 +237,19 @@ def _build_params_dict(self, weight, bias, **kwargs):
     def _build_params_dict_single(self, weight, bias, **kwargs):
         return [dict(params=bias, **kwargs)]
 
+    def test_save_optim_defaults(self):
+        weight = torch.randn(2, 2)
+        default = torch.randn(2)
+        opt = optim.Optimizer([weight], {'test_default': default})
+        filename = 'test_save_optim.pth'
+        torch.save(opt, filename)
+
+        if os.path.exists(filename):
+            loaded_opt = torch.load(filename)
+            self.assertEqual(loaded_opt.param_groups[0]['params'][0], weight)
+            self.assertEqual(loaded_opt.defaults['test_default'], default)
+            os.remove(filename)
```

Comment:
If any assert fails you are not cleaning up file. Consider using try, finally blocks.